state/core/src/main/java/com/expediagroup/streamplatform/streamregistry/state/StateValue.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2018-2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.state;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity;
+
+@EqualsAndHashCode
+@ToString
+@AllArgsConstructor
+public class StateValue {
```

Comment:
Is it not simpler to just add `boolean deleted` directly onto `Entity` (genuine question)?

Reply:
So there was a couple of reasons I didn't go for that approach:

1. We'd then expose another way of checking/retrieving deleted `Entity`s. Since we made this backwards compatible, the `EntityView::all` only returns `Entities` that aren't deleted. If there was a `deleted` flag on entity it might encourage users to query that value when its not needed.
2. When filter for deleted Entities, with this approach we only need to check the `deleted` flag on `StateValue`. If we added the flag to the entity there are two states where an entity is deleted (when `Entity::deleted == true` and when the entity itself is `null`).
3. I'm lazy 😛 . Since `Entity` is an interface, we'd need to implement `deleted` in all implementations of it. Obviously we could add something to the class hierarchy to handle this for us, but that just feels like more complication.

Does that make sense?